base/compiler/typelimits.jl

```diff
@@ -87,7 +93,7 @@ function _limit_type_size(@nospecialize(t), @nospecialize(c), sources::SimpleVec
         return t # fast path: unparameterized are always simple
     else
         ut = unwrap_unionall(t)
-        if isa(ut, DataType) && isa(c, Type) && c !== Union{} && c <: t
+        if isa(ut, DataType) && !has_free_typevars(c) && c !== Union{} && c <: t
             # TODO: need to check that the UnionAll bounds on t are limited enough too
             return t # t is already wider than the comparison in the type lattice
```

Comment:
Ideally we might just delete this case, since it is not sound without an additional check that `t` qualifies as issimplertype than `c` (hence the TODO, from before the issimplertype predicate existed). But that makes it more complex, while I am not entirely sure that this case is actually worth having. Any objection to trying just to delete this?

Reply:
n.b. there is also an identical bit of code in `type_more_complex` that should be kept in sync with any changes we make here

Reply:
We should find out if anybody depends on it.

Reply:
(i.e. I'm fine removing it and PkgEval will tell us).